src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ComCaseloadService.kt

```diff
@@ -187,6 +179,17 @@ class ComCaseloadService(
     }
   }
 
+  fun mapCaseToVaryLicence(cases: List<ManagedOffenderCrn>): Map<ManagedOffenderCrn, LicenceSummary> {
+    val licences = findExistingActiveAndVariationLicences(cases.mapNotNull { it.crn })
+    val casesToLicences = cases.associateWith { case -> findVaryLicenceToDisplay(licences.filter { licence -> case.crn == licence.crn }) }
+    return casesToLicences.mapNotNull { (case, licence) ->
```

Comment:
don't think you need to do the `associateWith` to build a map to then turn back to a list and then back to a map, 

could you combine both into a single mapNotNull?  

```kt
cases.mapNotNull { case -> 
    val caseLicences = licences.filter { licence -> case.crn == licence.crn }
    val varyLicence = findVaryLicenceToDisplay(caseLicences)
    when {
       varyLicence == null -> null
       else -> case to licence
    }
 }.toMap()
```

Reply:
or even
```kt
val licencesByCrn = licences.associateBy { it.crn }
cases.mapNotNull { case -> 
    val varyLicence = findVaryLicenceToDisplay(licencesByCrn[case.crn])
    when {
       varyLicence == null -> null
       else -> case to licence
    }
 }.toMap()
```

Reply:
Yep, completely agree
This is a great example of where I've progressively built up code and not gone back to check for efficient refactors!

Thanks Andy